### ISSUE TITLE
pycdf fails if CDF_LIB is not set

### ISSUE THREAD

**Issue by aburrell** [spacepy/spacepy#299]
My issue is that, when installing spacepy to use the cdf reading capability on Travis-CI and AppVeyor, SpacePy will fail unless CDF_LIB is set.

### Minimal example to reproduce issue:
Not sure how you would reproduce this locally, but the logs for failed tests are available at these two links:

https://ci.appveyor.com/project/aburrell/ocbpy/builds/32697607/job/j8nq3kf3imnj9y8e
(see line 957)

https://travis-ci.org/github/aburrell/ocbpy/jobs/681538148
(see line 1791)

This was fixed in Travis, by adding a CDF_LIB definition to the .travis.yml file.  I'm having a hard time finding the installation location in AppVeyor, so not sure how to fix it there yet.

### OS, Python version, and dependency version information:
OS: linux and windows
Python version: 2.7

### Version of SpacePy
Using pip to install spacepy, it is downloading version 0.2.1 on Travis and AppVeyor

**Comment by jtniehof** [comment]
The "Checked ." means it didn't find anything that looked remotely like a possible CDF library. Can you, on both platforms, let me know what you set CDF_LIB to in order to make it work? That'll give me an idea of if it's just adding another location to the magic checks, or if the magic checks themselves aren't working.

CDF_LIB is the most reliable way but it does try to check for anything "reasonable."

**Comment by aburrell** [comment]
On Travis, I set `CDF_LIB=$HOME/lib` and then it worked.  Still working on AppVeyor.

**Comment by jtniehof** [comment]
Is ~/lib in any of your library paths? (e.g. can you compile and/or run something linked against libcdf without specifying ~lib?) If so, things should be found and I need to fix that code. Otherwise I can probably add ~/lib (and ~/.local/lib) to the magic list, but since CDF doesn't install there by default (it by default installs into the source tree, oddly enough) it's not currently on there.

Basically I try to use normal system mechanisms for determining where the user wants libraries to be searched, and then fall back to rooting around where the CDF might have dropped it if that fails. I'm a little wary of doing too much "dig all over the system for anything that looks like a CDF library" because that can lead to unexpected results.

**Comment by aburrell** [comment]
Well, since these are CI systems and I am not a power user (as made obvious by my long debugging struggle!), I can't answer that question.  There is nothing else I am using in my CI tested routines that uses libcdf.

I am not convinced this is a bug in pyCDF, but it would be a useful addition to the documentation (to tell the user that they need to set the CDF_LIB parameter, probably to a certain value, if they have are using pyCDF as a dependency in their unit testing on an external system).

> I'm a little wary of doing too much "dig all over the system for anything that looks like a CDF library" because that can lead to unexpected results.
I entirely agree with you on that count.

> Otherwise I can probably add ~/lib (and ~/.local/lib) to the magic list
Well, this is probably not the worst idea, since they are standard library directories.  But I think just adding a paragraph to the documentation may be the more elegant solution.

**Comment by jtniehof** [comment]
> I am not convinced this is a bug in pyCDF, but it would be a useful addition to the documentation (to tell the user that they need to set the CDF_LIB parameter, probably to a certain value, if they have are using pyCDF as a dependency in their unit testing on an external system).

Check starting from the third paragraph of https://spacepy.github.io/pycdf.html ; PR and/or suggestiosn welcome for changes there.

**Comment by jtniehof** [comment]
I'm opening this back up because I do plan to add more search paths on Linux. I also should get a Windows 10 VM set up and see if the default location has changed for the CDF library. If nothing else I should probably be using %SYSTEMDRIVE% rather than hard-coded C:; that might help.
